apps/extension/include/subndarray.h

```diff
@@ -0,0 +1,36 @@
+#ifndef TVM_SUBNDARRAY_H_
+#define TVM_SUBNDARRAY_H_
+
+#include <tvm/runtime/packed_func.h>
+#include <tvm/runtime/ndarray.h>
+
+namespace subndarray {
+class NDSubClass;
+} // namespace subndarray
+
+namespace tvm {
+namespace runtime {
+template<>
+struct array_type_index<subndarray::NDSubClass> {
+  static constexpr uint32_t code = 1;
+};
+} // namespace runtime
+} // namespace tvm
+
+namespace subndarray {
+
+class NDSubClass : public tvm::runtime::NDArray {
+public:
+  class Container;
+  NDSubClass();
+  NDSubClass(NDArray::Container *data);
+}; // class NDSubClass
+
+class NDSubClass::Container : public NDArray::Container {
+public:
+  Container();
```

Comment:
subclass container need to initialize the array_type_index_ to be the indicated value